frontend/src/metabase/dashboard/components/DashboardInfoSidebar.tsx

```diff
@@ -47,17 +50,28 @@ const DashboardInfoSidebar = ({
     PLUGIN_CACHING.isEnabled() && MetabaseSettings.get("enable-query-caching");
 
   const handleDescriptionChange = useCallback(
-    async (description: string) => {
-      await setDashboardAttribute("description", description);
+    (description: string) => {
+      setDashboardAttribute("description", description);
       saveDashboardAndCards(dashboard.id);
     },
-    [setDashboardAttribute, saveDashboardAndCards, dashboard],
+    [dashboard.id, saveDashboardAndCards, setDashboardAttribute],
   );
 
-  const handleUpdateCacheTTL = async (cache_ttl: number | null) => {
-    await setDashboardAttribute("cache_ttl", cache_ttl);
-    saveDashboardAndCards(dashboard.id);
-  };
+  const handleUpdateCacheTTL = useCallback(
+    (cache_ttl: number | null) => {
+      setDashboardAttribute("cache_ttl", cache_ttl);
+      saveDashboardAndCards(dashboard.id);
+    },
+    [dashboard.id, saveDashboardAndCards, setDashboardAttribute],
+  );
+
+  const handleToggleAutoApplyFilters = useCallback(
```

Comment:
This function will also be used as another hook's dependency, so I need to memoize it.
https://github.com/metabase/metabase/blob/3fb962b9b816112a33f94079b63abe862dc7fe3f/frontend/src/metabase/core/components/Toggle/Toggle.tsx#L23-L28